### PR TITLE
extract test-utils as its own package for use with plugins, etc.

### PR DESCRIPTION
**Pull request by fastfrwrd**
## Hey, I just made a Pull Request!

I found myself wanting to use the testUtils in the plugin tests, so I've moved it to its own package in this PR. I opted to make it its own package because I feel core shouldn't have peer dependencies on the testing stuff.

*Note: this introduces a dependency cycle that should be resolved*. The issue is that the test-utils depends on the theme and ErrorBoundary from core, and core depends on this project. I'm not really sure what the best path forward is for that part... The theme could be its own package pretty cleanly (imagining a world where you may roll your own theme and inject it as a package...), and then maybe just a custom error boundary inside `@backstage/test-utils` to drop the core dep.

#### :heavy_check_mark: Checklist

- [x] All tests are passing `yarn test`
- [x] Relevant documentation updated
- [x] Prettier run on changed files

### CODE REVIEW

File: packages/core/src/layout/Header/Header.tsx [hunk 1]
```diff
@@ -17,12 +17,13 @@
...
 import { Typography, Tooltip, makeStyles } from '@material-ui/core';
+import { BackstageTheme } from '@backstage/theme';
+
 import { Theme } from '../Page/Page';
 // import { Link } from 'shared/components';
-import { BackstageTheme } from '../../theme/theme';
 import Waves from './Waves';
 
-const useStyles = makeStyles<BackstageTheme>(theme => ({
+const useStyles = makeStyles<typeof BackstageTheme>(theme => ({
```
Comment (fastfrwrd) [inline packages/core/src/layout/Header/Header.tsx:26]: the good news in this PR: you can use Typescripts' `typeof` to get a nice little bit of streamlined behavior here.

File: packages/theme/src/BackstageTheme.ts [hunk 2]
```diff
@@ -66,6 +69,7 @@ const extendedThemeConfig = {
         default: COLORS.DEFAULT_PAGE_THEME_COLOR,
       },
     },
+    // @ts-ignore
```
Comment (fastfrwrd) [inline packages/theme/src/BackstageTheme.ts:72]: the bad news in this PR: making the types work in this file was proving annoying and I decided to nope out on the last few cases where it seems like some annotations are needed to convince the type system that this is valid.

File: packages/cli/config/eslint.js [hunk 3]
```diff
@@ -61,7 +61,7 @@ module.exports = {
       },
     },
     {
-      files: ['**/*.test.*', '**/src/setupTests.*', '**/src/testUtils/**'],
```
Comment (Rugvip) [inline packages/cli/config/eslint.js:64]: 👍

File: packages/cli/src/commands/plugin/rollup.config.ts [hunk 4]
```diff
@@ -46,7 +46,6 @@ export default {
     json(),
     typescript({
       include: `${paths.resolveTarget('src')}/**/*.{js,jsx,ts,tsx}`,
-      exclude: [paths.resolveTarget('src/testUtils/**')],
```
Comment (Rugvip) [inline packages/cli/src/commands/plugin/rollup.config.ts:49]: 👍

File: packages/core/package.json [hunk 5]
```diff
@@ -43,9 +43,14 @@
   },
   "devDependencies": {
     "@backstage/cli": "^0.1.1-alpha.1",
+    "@backstage/test-utils": "^0.1.1-alpha.1",
+    "@backstage/theme": "^0.1.1-alpha.1",
```
Comment (Rugvip) [inline packages/core/package.json:47]: just slap it in deps once imo.

File: packages/core/package.json [hunk 6]
```diff
@@ -43,9 +43,14 @@
...
     "@testing-library/jest-dom": "^4.2.4",
     "@testing-library/react": "^9.3.2",
     "@testing-library/user-event": "^7.1.2",
     "react-router": "^5.1.2"
+  },
+  "peerDependencies": {
+    "@backstage/theme": "^0.1.1-alpha.1"
```
Comment (Rugvip) [inline packages/core/package.json:54]: we'll likely move away from peer deps, since it doesn't work well with workspaces

**Review by Rugvip — DISMISSED** [review]
:shipit:

Btw, I think `@backstage/test-utils` is the place to put mock implementations of core APIs, which is currently an open questions related to APIs :grin:

**Comment by fastfrwrd** [comment]
OK, I think everything from the PR is addressed and ready. @Rugvip it dismissed your review when I pushed up but I think we should be good.

**Review by Rugvip — APPROVED** [review]

**Comment by Rugvip** [comment]
🚢 💨

**Comment by fastfrwrd** [comment]
WOOOO

**Pull request merged**